project/FrontEnd/collaborative_science_platform/lib/screens/workspace_page/mobile_workspace_page/widget/entry_card.dart

```diff
@@ -50,146 +50,164 @@ class _EntryCardState extends State<EntryCard> {
       crossAxisAlignment: CrossAxisAlignment.center,
       children: [
         Text(
-          widget.entry.isTheoremEntry ? "Theorem"
-          : widget.entry.isProofEntry ? "Proof"
-          : "",
+          widget.entry.isTheoremEntry
+              ? "Theorem"
+              : widget.entry.isProofEntry
+                  ? "Proof"
+                  : "",
           style: TextStyle(
             fontWeight: FontWeight.w600,
             fontSize: 16.0,
-            color: widget.entry.isTheoremEntry ? Colors.green.shade800
-              : widget.entry.isProofEntry ? Colors.yellow.shade800
-              : Colors.blue.shade800,
+            color: widget.entry.isTheoremEntry
+                ? Colors.green.shade800
+                : widget.entry.isProofEntry
+                    ? Colors.yellow.shade800
+                    : Colors.blue.shade800,
           ),
         ),
-        if (widget.entry.isFinalEntry) Text(
-          " (Final)",
-          style: TextStyle(
-            fontWeight: FontWeight.w600,
-            fontSize: 16.0,
-            color: Colors.red.shade800,
+        if (widget.entry.isFinalEntry)
+          Text(
+            " (Final)",
+            style: TextStyle(
+              fontWeight: FontWeight.w600,
+              fontSize: 16.0,
+              color: Colors.red.shade800,
+            ),
           ),
-        ),
       ],
     );
   }
 
   Widget iconRow(bool pending) {
-    return !pending ? Row(
-      crossAxisAlignment: CrossAxisAlignment.center,
-      children: [
-        readOnly ? IconButton(
-          onPressed: () { // Make editable
-            setState(() {
-              readOnly = false;
-            });
-          },
-          icon: const Icon(Icons.edit),
-        ) : Row(
-          children: [
-            IconButton(
-              onPressed: () {
-                // Save the changes
-                setState(() {
-                  readOnly = true;
-                });
-              },
-              icon: const Icon(Icons.check),
-            ),
-            IconButton(
-              onPressed: () {
-                // Do not save the changes
-                setState(() {
-                  readOnly = true;
-                });
-              },
-              icon: const Icon(Icons.close),
-            ),
-          ],
-        ),
-        IconButton(
-          onPressed: () {
-            showDialog(
-              context: context,
-              builder: (context) =>
-                AppAlertDialog(
-                  text: "Do you want to delete the entry?",
-                  actions: [
-                    AppButton(
-                      text: "Yes",
-                      height: 40,
-                      onTap: () {
-                        setState(() { // delete the entry
-                          widget.onDelete();
+    return !pending
```

Comment:
Same comment goes here. It is challenging to follow what's going on here even the case is simple. We may create pendingWidget, entryWidget in the same file if necessary.

Since it is still not that complex widget tree, we may omit for now.